standard/standard-library.md

```diff
@@ -379,18 +379,44 @@ The following types, including the members listed, shall be defined in a conform
 
 A conforming implementation may provide `Task.GetAwaiter()` and `Task<TResult>.GetAwaiter()` as extension methods.
 
+> Note to TG2 reviewers: Required vs. Optional library type members: We need to indicate which members of `Index` and `Range` are required and which are optional.
+
 ```csharp
 namespace System
 {
     public class FormattableString : IFormattable { }
 
+    public readonly struct Index : IEquatable<Index>
+    {
+        public Index(int value, bool fromEnd = false);
+        public int Value { get; }
+        public bool Equals(Index other);
+        public override bool Equals(object? value);
+        public override int GetHashCode();
+        public int GetOffset(int length);
+        public override string ToString();
+        public static implicit operator Index(int value);
+    }
+
     public class OperationCanceledException : Exception
     {
         public OperationCanceledException();
         public OperationCanceledException(string? message);
         public OperationCanceledException(string? message, Exception? innerException);
     }
 
+    public struct Range : IEquatable<Range>
+    {
+        public Range (Index start, Index end);
+        public Index End { get; }
+        public Index Start { get; }
+        public override bool Equals(object? value);
+        public bool Equal (Range other);
+        public override int GetHashCode();
+        public (int,int) GetOffsetAndLength(int length);
```

Comment:
Should we mention the tuple element names that the BCL shipped?
```suggestion
        public (int Offset, int Length) GetOffsetAndLength(int length);
```
(<https://learn.microsoft.com/en-us/dotnet/api/system.range.getoffsetandlength?view=netcore-3.0>)